src/transformers/pipelines.py

```diff
@@ -408,7 +497,7 @@ def __init__(
     ):
 
         if framework is None:
-            framework = get_framework()
+            framework = get_framework(model)
```

Comment:
This is what the `pipeline` function does, seemed good to do it too, in case a user is directly using the subclasses.